pkg/clusteragent/autoscaling/workload/external/recommender.go

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package external
+
+import (
+	"context"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	le "github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	pollingInterval              = 30 * time.Second
+	externalRecommenderID string = "extr"
+)
+
+// Recommender is the interface used to fetch external recommendations
+type Recommender struct {
+	recommenderClient *recommenderClient
+	store             *autoscaling.Store[model.PodAutoscalerInternal]
+}
+
+// NewRecommender creates a new Recommender to start fetching external recommendations
+func NewRecommender(podWatcher workload.PodWatcher, store *autoscaling.Store[model.PodAutoscalerInternal]) *Recommender {
+	recommenderClient := newRecommenderClient(podWatcher)
+
+	return &Recommender{
+		recommenderClient: recommenderClient,
+		store:             store,
+	}
+}
+
+// Run starts the Recommender interface to generate external recommendations
+func (r *Recommender) Run(ctx context.Context) {
+	log.Debugf("Starting external autoscaling recommender")
+	ticker := time.NewTicker(pollingInterval)
+
+	go func() {
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ticker.C:
+				r.process(ctx)
+			case <-ctx.Done():
+				log.Debugf("Stopping external autoscaling recommender")
+				return
+			}
+		}
+	}()
+}
+
+func (r *Recommender) process(ctx context.Context) {
+	// Filter pod autoscalers to only retrieve autoscalers where external recommender is enabled
+	externalRecommendationFilter := func(dpa model.PodAutoscalerInternal) bool {
+		return dpa.CustomRecommenderConfiguration() != nil
+	}
+	podAutoscalers := r.store.GetFiltered(externalRecommendationFilter)
+
+	log.Debugf("Found %d pod autoscalers with external recommender enabled", len(podAutoscalers))
+
+	for _, podAutoscaler := range podAutoscalers {
+		// Fetch external recommendations
+		recommendation, err := r.recommenderClient.GetReplicaRecommendation(ctx, podAutoscaler)
```

Comment:
With a single goroutine and a `10s` timeout, we could get behind relatively quickly if one recommender is down. We should probably revisit that and add proper handling

Reply:
I'll make a card to track this + other pending improvements!